dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt

```diff
@@ -165,6 +177,9 @@ internal abstract class PluginFeaturesService : BuildService<PluginFeaturesServi
         private const val V2_PLUGIN_NO_WARN_FLAG_PRETTY =
             "$V2_PLUGIN_ENABLED_FLAG.noWarn"
 
+        private const val V2_PLUGIN_MIGRATION_HELPERS_FLAG =
+            "org.jetbrains.dokka.experimental.gradlePlugin.enableV2MigrationHelpers"
```

Comment:
Good point. Maybe we should just have two flags:

- `org.jetbrains.dokka.experimental.gradlePlugin.enableV2`
- `org.jetbrains.dokka.experimental.gradlePlugin.migrateToV2` - enables v2, and logs the warning, and enables these migration helpers, and anything else that we come up with the the future.

The migration guide can say "start with `migrateToV2=true`, then when you're done, change it to `enableV2` to disable the warning."

Reply:
Yeah, nice! That sounds better I think. Let's wait for @Tapchicoma review before proceeding, may be there are other options/opinions here.

Also, just an idea, may be `enableV2` and `enableV2Migration` would be nicer, so that it's easier to change from one to another :)

Reply:
Sure, sounds good. 

If we all agree then I suggest making this change in a new PR and keeping this one as-is.

Reply:
Or a single property with different values?

```properties
# Initial migration mode. Enable migration helpers, logs a warning.
org.jetbrains.dokka.experimental.gradlePlugin.enableV2=migration

# After migration. Removes migration helpers, disables warnings
org.jetbrains.dokka.experimental.gradlePlugin.enableV2=true
```

Reply:
or even just `mode` and then have all possible states as values:
- `v1enabled`
- `v2enabledWithHelpers`
- `v2Enabled`
- `v2EnabledNoWarnings`

Reply:
https://youtrack.jetbrains.com/issue/KT-71236/Rationalise-flags-for-configuring-DGPv1-and-DGPv2-modes

Reply:
https://youtrack.jetbrains.com/issue/KT-71236/Rationalise-flags-for-configuring-DGPv1-and-DGPv2-modes

Reply:
Agree, let's do this with one `mode` property with those four states